tests/integration/conftest.py

```diff
@@ -780,21 +778,212 @@ def external_hostname_fixture() -> str:
+        f"{oathkeeper.name}:certificates", "self-signed-certificates"
+    )
+    await application.model.add_relation(
+        "traefik-public:receive-ca-cert", "self-signed-certificates"
+    )
+    await application.model.applications["traefik-public"].set_config(
+        {"enable_experimental_forward_auth": "True"}
+    )
+    await application.model.add_relation(
+        f"{oathkeeper.name}", "traefik-public:experimental-forward-auth"
+    )
+    await application.model.add_relation(f"{oathkeeper.name}:kratos-info", "kratos")
+    # Needed per https://github.com/canonical/oathkeeper-operator/issues/49
+    await application.model.applications["kratos"].set_config({"dev": "True"})
+    await application.model.add_relation(f"{application.name}:ingress", "traefik-public")
+    await application.model.add_relation(f"{application.name}:auth-proxy", oathkeeper.name)
+
+    await application.model.wait_for_idle(
+        status="active",
+        apps=[application.name, oathkeeper.name] + [app.name for app in identity_platform],
+        raise_on_error=False,
+        timeout=30 * 60,
+        idle_period=5,
+    )
+
+    get_redirect_uri_action = (
+        await application.model.applications["kratos-external-idp-integrator"]
+        .units[0]
+        .run_action("get-redirect-uri")
+    )
+    action_output = await get_redirect_uri_action.wait()
+    update_redirect_uri(client, action_output.results["redirect-uri"])
+    return oathkeeper
+
+
+@pytest.fixture(scope="session", name="client")
+def client_fixture() -> Client:
+    """k8s client."""
+    return Client(config=KubeConfig.from_file(KUBECONFIG), field_manager="dex-test")
+
+
+@pytest.fixture(scope="module", name="ext_idp_service")
+def ext_idp_service_fixture(ops_test: OpsTest, client: Client) -> Generator[str, None, None]:
+    """Deploy a DEX service on top of k8s for authentication."""
+    try:
+        create_dex_resources(client)
+        # We need to set the dex issuer_url to be the IP that was assigned to
+        # the dex service by metallb. We can't know that before hand, so we
+        # reapply the dex manifests.
+        apply_dex_resources(client)
+        yield get_dex_service_url(client)
+    finally:
+        if not ops_test.keep_model:
+            for obj in get_dex_manifest():
+                try:
+                    # mypy doesn't work well with lightkube
+                    client.delete(
+                        type(obj),
+                        obj.metadata.name,  # type: ignore
+                        namespace=obj.metadata.namespace,  # type: ignore
+                    )
+                except ApiError:
+                    pass
+
+
+@pytest.fixture()
+def external_user_email() -> str:
+    """Username for testing proxy authentication."""
+    return "admin@example.com"
+
+
+@pytest.fixture()
+def external_user_password() -> str:
+    """Password for testing proxy authentication."""
+    return "password"
```

Comment:
Should we do a randomised hex string here from the `secrets` module?

Reply:
It would be a bit more complex that that, since I'd have to pass the bcript hashed password all the way down to the dex k8s descriptors. It is definitely possible, though, if you think that's worth it